executor.go

```diff
@@ -0,0 +1,139 @@
+package goptions
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+)
+
+var (
+	//ErrNoCommand This should never happen.
+	ErrNoCommand = errors.New("No command specified")
+)
+
+//Executer used to allow auto execution of commands.
+type Executer interface {
```

Comment:
The Executer _needs_ proper documentation. I have no idea how to use this or what it does honestly. I’d like a (testable) example here.

I am gonna review the rest of this without really understanding what this is supposed to do. Apologies if I misunderstand something.